Rhythm/Interpolation.swift

```diff
@@ -0,0 +1,193 @@
+//
+//  Tempo.Interpolation.swift
+//  Rhythm
+//
+//  Created by James Bean on 5/23/17.
+//
+//
+
+import Foundation
+import ArithmeticTools
+
+/// Interpolation between two `Tempo` values.
+///
+/// - TODO: Consider defining multiple interpolation types (linear, exponential, etc.)
+///
+public struct Interpolation {
+
+    // MARK: - Associated Types
+    public enum EasingError: Error {
+        case badValue(String, Double)
+    }
+
+    /// Easing of `Interpolation`.
+    public enum Easing {
+
+        /// Linear interpolation.
+        case linear
+
+        /// Exponential interpolation in, with the given `exponent`.
+        case exponentialIn(exponent: Double)
+
+        /// Exponential interpolation in-out, with the given `exponent`.
+        case exponentialInOut(exponent: Double)
+
+        /// Ease in / ease out (half sine wave)
+        case sineInOut
+
+        // Custom timing function modeled with cubic Bézier curve control points in the
+        // form (x,y)
+        case custom(controlPoint1: (Double, Double), controlPoint2: (Double, Double))
+
+        /// - returns: The easing function evaluated at `x`.
+        ///
+        func evaluate(at x: Double) throws -> Double {
+            guard x >= 0 && x <= 1 else {
```

Comment:
For some reason, I've adopted a style of putting a newline before guard statements, to really isolate them on their own island.